lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -189,22 +191,11 @@ getBlock = label "getBlock" $ do
     return $ Block header msgs
 
 -- | Extract a 'Block' id from a serialized 'Block'.
---
--- FIXME:
--- Note that, this is a very simplistic way of doing it (although rather
--- efficient) and a more correct one would required:
---
--- - Unserializing a block
--- - Extracting its header
--- - Serializing the header
--- - Computing the hash of this serialized header
---
--- This is quite some busy work but we should come to that eventually.
 getBlockId :: Get (Hash "BlockHeader")
-getBlockId = do
+getBlockId = lookAhead getBlock *> (label "getBlockId" $ do
```

Comment:
Didn't thought of it initially but, we don't have to actually unserialize and re-serialize the block. We can simply "lookAhead". If this fails, we know that we haven't a block and the rest is irrelevant. Hence I removed the note.